apps/snitch_core/lib/core/data/model/tax_rate.ex

```diff
@@ -0,0 +1,100 @@
+defmodule Snitch.Data.Model.TaxRate do
+  @moduledoc """
+  Model functions TaxRate.
+  """
+
+  use Snitch.Data.Model
+  alias Snitch.Data.Schema.TaxRate
+
+  @doc """
+  Creates a TaxRate with supplied `params`.
+  """
+  @spec create(map) :: {:ok, TaxRate.t()} | {:error, Ecto.Changeset.t()}
+  def create(params) do
+    params =
+      if Map.has_key?(params, :calculator) do
+        %{calculator: calc} = params
+        %{params | calculator: String.to_atom(calc)}
```

Comment:
Since the schema declares `:calculator` to be an `atom`, why would anyone pass a `string` here? In fact, you should never convert user-supplied strings into atoms! Passing a string should be strictly forbidden.